addons/pinniped/post-deploy/pkg/pinnipedclientset/pinnipedclientset.go

```diff
@@ -0,0 +1,92 @@
+// Copyright 2021 VMware, Inc. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+// Package pinnipedclientset provides implementations for clientsets for the Pinniped API.
+//
+// These hand-written Pinniped clientset implementations are required because the Pinniped instance
+// running on TKG uses a non-default API group, and therefore we must use a dynamic (non-generated)
+// client to talk to it.
+package pinnipedclientset
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/vmware-tanzu/tanzu-framework/addons/pinniped/post-deploy/pkg/constants"
+	"go.uber.org/zap"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/dynamic"
+)
+
+// translateAPIGroup translates the provided pinniped baseAPIGroup to a new API group based on the
+// provided newAPIGroupSuffix.
+//
+// We assume that all baseAPIGroup's will end in "pinniped.dev", and therefore we can safely replace
+// the reference to "pinniped.dev" with the provided newAPIGroupSuffix. If the provided baseAPIGroup
+// does not end in "pinniped.dev", then this function will return an empty string.
+func translateAPIGroup(baseAPIGroup, newAPIGroupSuffix string) string {
```

Comment:
Should we add a test for this path (when `baseAPIGroup` doesn't contain `pinniped.dev`)?